src/clojure/contrib/util/time_convert.cljc

```diff
@@ -0,0 +1,77 @@
+(ns clojure.contrib.util.time-convert
```

Comment:
I think a case could be made that all of this time converting code could be in it's own independent library. Let me know if you would prefer that approach and I can extract this and require as a dependency.

Currently on the `master` branch the `datetime` function assumes that an `org.joda.time.DateTime` is passed in, and I did not want to break that contract with this change. It seemed natural to add a few more obvious date time representations here as well, so with this PR the `datetime` function is more flexible than before.

Reply:
At this point, namespaces have been changed, and the Maven coordinates are changed, and the version number is still alpha, so just doing it *right* is on the table.

Reply:
Ok - great! It would definitely simplify this PR if the `datetime` function could always assume that a java-time object is passed in.